resources/idlharness.js

```diff
@@ -84,10 +84,26 @@ function minOverloadLength(overloads)
     .reduce(function(m, n) { return Math.min(m, n); });
 }
 
+// A helper to get the global of a Function object.  This is needed to determine
+// which global exceptions the function throws will come from.
+function globalOf(func)
+{
+    var global;
+    try {
+        // Use the fact that .constructor for a Function object is the Function
+        // constructor, which can be used to mint a new function in the right
+        // global.
+        global = func.constructor("return this;")();
+    } catch (e) {
+        // If it fails, just fall back to 'self'.
```

Comment:
Would be useful to explain what it means if this fails. Was 'func' not a function then?

Reply:
Or a function with a weird `.constructor` defined on it, or with a weird proto chain....  I can add a comment,